lib/sms/sms.c

```diff
@@ -241,7 +244,7 @@ static int sms_init(void)
 	}
 	/* Parameters 1 to 4 should be 0 if no client is registered. */
 	if (cnmi_value1 != 0 || cnmi_value2 != 0 || cnmi_value3 != 0 || cnmi_value4 != 0) {
-		LOG_ERR("Only one SMS client can be registered");
+		LOG_WRN("Only one SMS client can be registered");
```

Comment:
You could check here if `sms_client_registered` is already `true`, in which case it's expected that +CNMI value is not `0,0,0,0` and the function could return `0`. Otherwise print an error and return `-EBUSY`. That would also simplify the condition on line 396.

Reply:
I made this change and added a debug log when the registration is in place so we know what happens.
I'll still add some unit test.